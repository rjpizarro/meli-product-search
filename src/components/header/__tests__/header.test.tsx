import React from 'react';
import renderer from 'react-test-renderer';
import {fireEvent, render} from '@testing-library/react';
import Header from '../index'

const mockOnClick = jest.fn()

jest.mock('../../../components/logo', () => (props) => <div />)
jest.mock('../../../components/base-button', () =>
    (props) => <button data-testid="base-button" onClick={(() => props.onClick())} />
)

describe('Header', () => {
    it('should render correctly ', () => {
        const tree = renderer
            .create(
            <Header onLogoClick={mockOnClick} >
                <div>children</div>
            </Header>
        ).toJSON()

        const { getByTestId }  = render(
            <Header onLogoClick={mockOnClick} >
                <div data-testid="child-component">children</div>
            </Header>
        );

        const button = getByTestId('base-button')

        fireEvent.click(button)

        expect(tree).toMatchSnapshot()
        expect(mockOnClick).toHaveBeenCalledTimes(1)
        expect(getByTestId('child-component')).toBeInTheDocument()
    })
})

