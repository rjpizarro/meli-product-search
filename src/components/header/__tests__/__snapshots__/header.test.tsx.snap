// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly  1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="row center-xs"
    >
      <div
        className="col-xs-3 col-md-1 header__logo"
      >
        <button
          data-testid="base-button"
          onClick={[Function]}
        />
      </div>
      <div
        className="col-xs-10 col-md-9"
      >
        <div>
          children
        </div>
      </div>
    </div>
  </div>
</header>
`;
